tests/unit/test_synapse_workload.py

```diff
@@ -74,29 +71,15 @@ def test_enable_metrics_error(monkeypatch: pytest.MonkeyPatch):
         synapse.enable_metrics(container_mock)
 
 
-def test_enable_saml_success():
+def test_enable_saml_success(harness_with_saml: Harness):
     """
     arrange: set mock container with file.
     act: change the configuration file.
     assert: new configuration file is pushed and SAML is enabled.
     """
     # This test was given as an example in this comment by Ben Hoyt.
     # https://github.com/canonical/synapse-operator/pull/19#discussion_r1302486670
-    # Arrange: set up harness and container filesystem
-    harness = Harness(SynapseCharm)
-    harness.update_config({"server_name": TEST_SERVER_NAME, "public_baseurl": TEST_SERVER_NAME})
-    relation_id = harness.add_relation("saml", "saml-integrator")
-    harness.add_relation_unit(relation_id, "saml-integrator/0")
-    metadata_url = "https://login.staging.ubuntu.com/saml/metadata"
-    harness.update_relation_data(
-        relation_id,
-        "saml-integrator",
-        {
-            "entity_id": "https://login.staging.ubuntu.com",
-            "metadata_url": metadata_url,
-        },
-    )
-    harness.set_can_connect(SYNAPSE_CONTAINER_NAME, True)
+    harness = harness_with_saml
```

Comment:
This test was supposed to be used as an example. Why it was changed?

Reply:
I reverted my changes.